### PR TITLE
Managed database cross subscription restore

### PR DESCRIPTION
**Pull request by milanbrkic-ms**
## Checklist

- [X] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Sql/Sql.Test/ScenarioTests/ManagedDatabaseCrudScenarioTests.ps1 [hunk 1]
```diff
@@ -468,4 +468,124 @@ function Test-SetManagedDatabase
...
+
+		$pointInTime = ([DateTime]$deletionDate).AddMinutes(-10)
+		
+		$targetSub = (Get-AzContext).Subscription.Id
+		$targetRg = Create-ResourceGroupForTest
+		$targetManagedInstance = Create-ManagedInstanceForTest $targetRg
+		$targetManagedDatabaseName = Get-ManagedDatabaseName
+
+		# Wait for mi to stabilize
+		if([Microsoft.Azure.Test.HttpRecorder.HttpMockServer]::Mode -eq "Record"){
```
Comment (ostojicb) [inline src/Sql/Sql.Test/ScenarioTests/ManagedDatabaseCrudScenarioTests.ps1:567]: same

File: src/Sql/Sql/ManagedDatabase/Cmdlet/RestoreAzureSqlManagedDatabase.cs [hunk 2]
```diff
@@ -441,22 +582,71 @@ protected override AzureSqlManagedDatabaseModel GetEntity()
...
+                    }
+                    model.CrossSubscriptionTargetManagedInstanceId = "/subscriptions/" + TargetSubscriptionId +
+                            "/resourceGroups/" + TargetResourceGroupName +
+                            "/providers/Microsoft.Sql/managedInstances/" + TargetInstanceName;
+                    break;
+                case PointInTimeCrossSubscriptionRestoreFromResourceIdParameterSet:
+                    var sourceResourceInfo = new ResourceIdentifier(ResourceId);
+                    SubscriptionId = sourceResourceInfo.Subscription;
+                    ResourceGroupName = sourceResourceInfo.ResourceGroupName;
+                    InstanceName = sourceResourceInfo.ParentResource.Split(new[] { '/' })[1];
```
Comment (ostojicb) [inline src/Sql/Sql/ManagedDatabase/Cmdlet/RestoreAzureSqlManagedDatabase.cs:623]:
```suggestion
                    InstanceName = sourceResourceInfo.ParentResource.Split(new[] { '/' }, StringSplitOptions.RemoveEmptyEntries).Last();
```
nit

Comment (milanbrkic-ms) [reply src/Sql/Sql/ManagedDatabase/Cmdlet/RestoreAzureSqlManagedDatabase.cs:623]: I don't think instance name would be `last` in this case, but `last -1`. SourceResourceInfo is info about database where type is `Microsoft.Sql/managedInstances/databases` or `Microsoft.Sql/managedInstances/restorableDroppedDatabases` if we are restoring deleted database.

Any suggestion in this case?

File: src/Sql/Sql/help/Restore-AzSqlInstanceDatabase.md [hunk 3]
```diff
@@ -403,6 +467,18 @@ Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
+```yaml
```
Comment (isra-fel) [inline src/Sql/Sql/help/Restore-AzSqlInstanceDatabase.md:470]: The heading for this parameter is missing

File: src/Sql/Sql.Sdk/README.md [hunk 4]
```diff
@@ -98,15 +98,15 @@ input-file:
...
   - https://raw.githubusercontent.com/Azure/azure-rest-api-specs/main/specification/sql/resource-manager/Microsoft.Sql/preview/2022-02-01-preview/ManagedDatabaseAdvancedThreatProtectionSettings.json
   - https://raw.githubusercontent.com/Azure/azure-rest-api-specs/main/specification/sql/resource-manager/Microsoft.Sql/preview/2022-02-01-preview/ManagedInstanceAdvancedThreatProtectionSettings.json
   - https://raw.githubusercontent.com/Azure/azure-rest-api-specs/main/specification/sql/resource-manager/Microsoft.Sql/preview/2022-02-01-preview/DatabaseAdvancedThreatProtectionSettings.json
   - https://raw.githubusercontent.com/Azure/azure-rest-api-specs/main/specification/sql/resource-manager/Microsoft.Sql/preview/2022-02-01-preview/ServerAdvancedThreatProtectionSettings.json
   - https://raw.githubusercontent.com/Azure/azure-rest-api-specs/main/specification/sql/resource-manager/Microsoft.Sql/preview/2021-11-01-preview/ManagedServerDnsAliases.json
+  - https://raw.githubusercontent.com/Azure/azure-rest-api-specs/main/specification/sql/resource-manager/Microsoft.Sql/preview/2022-05-01-preview/ManagedDatabases.json
```
Comment (isra-fel) [inline src/Sql/Sql.Sdk/README.md:108]: Use commit hash instead of branch name in the uri so that the end result is consistant.

**Review by isra-fel — CHANGES_REQUESTED** [review]
LGTM with 2 minor comments

File: src/Sql/Sql/help/Restore-AzSqlInstanceDatabase.md (hunk 3 above)
Comment (milanbrkic-ms) [reply src/Sql/Sql/help/Restore-AzSqlInstanceDatabase.md:470]: It is not missing, but some parameters are now required in curtain situations. In this case, subscriptionId is required for 2 parameter set, and for others are not. 
Also, I was using given documentation to re-generate md file.

**Comment by milanbrkic-ms** [comment]
@isra-fel I have addressed your comments. Please look PR again, and merge if everything is fine

**Review by isra-fel — APPROVED** [review]
LGTM

**Pull request merged**